staging/src/k8s.io/apiserver/pkg/storage/etcd3/watcher.go

```diff
@@ -376,6 +379,9 @@ func (wc *watchChan) prepareObjs(e *event) (curObj runtime.Object, oldObj runtim
 func decodeObj(codec runtime.Codec, versioner storage.Versioner, data []byte, rev int64) (runtime.Object, error) {
 	obj, err := runtime.Decode(codec, []byte(data))
 	if err != nil {
+		if fatalOnDecodeError {
+			panic(err) // we are running in a test environment and thus an error here is due to a coder mistake
```

Comment:
OK - I got the motivation now and that sounds fine to me.
In that case, I vote for consistency in all those mechanism and suggest using some env var to enable/disable it. For the mutation detector we have this:
https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/client-go/tools/cache/mutation_detector.go#L34